Plugins/Geant4/src/Geant4Converters.cpp

```diff
@@ -229,6 +230,72 @@ Acts::Geant4ShapeConverter::trapezoidBounds(const G4Trd& g4Trd) {
   return {std::move(tBounds), rAxes, thickness};
 }
 
+std::tuple<std::shared_ptr<Acts::TrapezoidBounds>, std::array<int, 2u>, double>
+Acts::Geant4ShapeConverter::trapezoidBounds(const G4Trap& g4Trap) {
+  // primary parameters
+  auto y1 = static_cast<double>(g4Trap.GetYHalfLength1());
+  auto y2 = static_cast<double>(g4Trap.GetYHalfLength2());
+  auto x1 = static_cast<double>(g4Trap.GetXHalfLength1());
+  auto x2 = static_cast<double>(g4Trap.GetXHalfLength2());
+  auto x3 = static_cast<double>(g4Trap.GetXHalfLength3());
+  auto x4 = static_cast<double>(g4Trap.GetXHalfLength4());
+  auto phi = static_cast<double>(g4Trap.GetPhi());
+  auto theta = static_cast<double>(g4Trap.GetTheta());
+  auto z = static_cast<double>(g4Trap.GetZHalfLength());
```

Comment:
I feel, for fundamental types, we usually try to avoid `auto`.

Reply:
That's actually a SonarCloud suggestion, since the static_cast makes the type explicit. 
I use here `static_cast`, as Geant4 has it's own `G4double` type, which we don't really have under control.

Reply:
https://github.com/acts-project/acts/pull/3494#discussion_r1710925101

Reply:
I see, right.